pkg/controller/scaledjob/scaledjob_controller.go

```diff
@@ -34,7 +44,12 @@ func Add(mgr manager.Manager) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileScaledJob{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+	return &ReconcileScaledJob{
+		client:                mgr.GetClient(),
+		scheme:                mgr.GetScheme(),
+		scaleLoopContexts:     &sync.Map{},
+		scaledJobsGenerations: &sync.Map{},
```

Comment:
I don't think these are used for the scaledJob controller, right? I only see `.Store()` and `.Delete()` called on `scaledJobsGenerations` and `.Load()`, and `.Delete()` on `scaleLoopContexts` which seems to make them no-op. In v2 `scaleLoopContext` is [encapsulated in `ScaleHandler`](https://github.com/kedacore/keda/blob/e107f26b4e50d31882af86eaf827467add4d3058/pkg/scaling/scale_handler.go#L81-L88). and the generation map is used for hpa updates in the scaled object controller and that doesn't apply here.

Reply:
I'll double check the v2 logic and confirm it. I'll update it in here.

Reply:
Hi @ahmelsayed 
I removed these logic and configuration.